### PR TITLE
Kinematics features

### PR DESCRIPTION
**Pull request by Lobotuerk**
Signed-off-by: Lobotuerk <jtlorente@ekumenlabs.com>

### CODE REVIEW

File: bullet/src/KinematicsFeatures.cc [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    assert(false);
+    return data;
+  }
+
+  const auto linkID = _id.ID();
+  const auto &linkInfo = this->links.at(linkID);
+  const auto modelIdentity = linkInfo->model;
+  // auto linkIndex = linkInfo->linkIndex;
+  const auto &modelInfo = this->models.at(modelIdentity);
+  const auto &model = modelInfo->model;
```
Comment (Blast545) [inline bullet/src/KinematicsFeatures.cc:28]: `modelInfo->model` does not exist anymore, probably you meant `linkInfo->model`.

File: bullet/src/KinematicsFeatures.cc [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  const btVector3 pos = model->getCenterOfMassTransform().getOrigin();
+  const btMatrix3x3 mat = model->getCenterOfMassTransform().getBasis();
+
+  auto eigenMat = convert(mat);
+  auto eigenVec = convert(pos);
+
+  data.pose.linear() = eigenMat;
+  data.pose.translation() = eigenVec;
+
+  // Get frame velocities
```
Comment (Blast545) [inline bullet/src/KinematicsFeatures.cc:39]: Those comments can be removed, as those are already in the previous implementation in the case we ever need them.

File: bullet/src/KinematicsFeatures.cc [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  if (_id.IsWorld())
+  {
+    ignerr << "Given a FrameID belonging to the world. This should not be "
+           << "possible! Please report this bug!\n";
+    assert(false);
+    return data;
+  }
+
+  const auto linkID = _id.ID();
+  const auto &linkInfo = this->links.at(linkID);
```
Comment (Blast545) [inline bullet/src/KinematicsFeatures.cc:24]: If we will the _id, which is a FrameID as a replacement to query the linkIDs I think we should something to exit the function cleanly in the case this function receives a not valid linkID.

I won't block this PR on that, but you can add a comment about it here so we remember later.

**Review by Blast545 — CHANGES_REQUESTED** [review]
Minor changes

File: bullet/src/KinematicsFeatures.cc (hunk 3 above)
Comment (Lobotuerk) [reply bullet/src/KinematicsFeatures.cc:30]: Added on 25225aa

File: bullet/src/KinematicsFeatures.cc (hunk 2 above)
Comment (Lobotuerk) [reply bullet/src/KinematicsFeatures.cc]: Removed on 25225aa

File: bullet/src/KinematicsFeatures.cc (hunk 1 above)
Comment (Lobotuerk) [reply bullet/src/KinematicsFeatures.cc]: Fixed on 25225aa

**Review by Blast545 — APPROVED** [review]

**Comment by Lobotuerk** [comment]
Merging

**Pull request merged**